src/CLIFramework/Extension/DaemonExtension.php

```diff
@@ -0,0 +1,202 @@
+<?php
+namespace CLIFramework\Extension;
+
+use CLIFramework\ServiceContainer;
+use CLIFramework\Command;
+use CLIFramework\Logger;
+use CLIFramework\Exception\ExtensionException;
+use CLIFramework\IO\StreamWriter;
+
+class DaemonExtension extends ExtensionBase
+{
+    private $config;
+    private $logger;
+    private $isNoClose = false;
```

Comment:
Alright. But it's better to provide an API to change the member than to make the member `protected`.